scripts/develop.sh first run

## Problem

Currently, after creating a new workspace in the dogfood instance - the `./scripts/develop.sh` script - allowing us to run coder in it - is not reliably working.

## Config

I created a simple workspace : 

- Not Nix instance (still happens with Nix instances too.)

## Logs

First try : 

```
go: downloading github.com/rcrowley/go-metrics v0.0.0-20201227073835-cf1acfcdf475
go: downloading github.com/go-ini/ini v1.67.0
go: downloading github.com/charmbracelet/lipgloss v1.0.0
go: downloading github.com/charmbracelet/x/ansi v0.4.5
../../../../rbac/object_gen.go:1:1: expected 'package', found 'EOF'
../../../rbac/object_gen.go:1:1: expected 'package', found 'EOF'
prog.go:15:2: no required module provides package github.com/coder/coder/v2/coderd/database/pubsub: go.mod file not found in current directory or any parent directory; see 'go help modules'
prog.go:13:2: no required module provides package go.uber.org/mock/mockgen/model: go.mod file not found in current directory or any parent directory; see 'go help modules'
2024/12/17 15:26:32 Loading input failed: exit status 1
coderd/database/pubsub/psmock/doc.go:4: running "mockgen": exit status 1
make: *** [Makefile:599: coderd/database/pubsub/psmock/psmock.go] Error 1
make: *** Waiting for unfinished jobs....
# github.com/coder/coder/v2/coderd/rbac
coderd/rbac/object_gen.go:1:1: syntax error: package statement must be first
make: *** [Makefile:587: coderd/database/dump.sql] Error 1
coder@vincentdev-broken:~/coder$ 
```

Second try : 

Faced this error - but still the process continued : 

```
go: downloading golang.org/x/tools v0.28.0
go: downloading github.com/dlclark/regexp2 v1.11.4
go: downloading github.com/go-sourcemap/sourcemap v2.1.3+incompatible
go: downloading golang.org/x/sync v0.10.0
generate
2024/12/17 15:29:54 ERROR parsing package, suggest running 'go get tailscale.com/derp' where calling the go generator to include the referenced package. error="/home/coder/coder/codersdk/healthsdk/healthsdk.go:11:2: could not import tailscale.com/derp (invalid package name: \"\")" pkg=github.com/coder/coder/v2/codersdk/healthsdk directory=github.com/coder/coder/v2/codersdk/healthsdk
2024/12/17 15:29:54 ERROR parsing package, suggest running 'go get tailscale.com/net/netcheck' where calling the go generator to include the referenced package. error="/home/coder/coder/codersdk/healthsdk/healthsdk.go:12:2: could not import tailscale.com/net/netcheck (invalid package name: \"\")" pkg=github.com/coder/coder/v2/codersdk/healthsdk directory=github.com/coder/coder/v2/codersdk/healthsdk
2024/12/17 15:29:54 ERROR parsing package, suggest running 'go get tailscale.com/tailcfg' where calling the go generator to include the referenced package. error="/home/coder/coder/codersdk/healthsdk/healthsdk.go:13:2: could not import tailscale.com/tailcfg (invalid package name: \"\")" pkg=github.com/coder/coder/v2/codersdk/healthsdk directory=github.com/coder/coder/v2/codersdk/healthsdk
2024/12/17 15:29:54 ERROR parsing package, suggest running 'go get tailscale.com/net/interfaces' where calling the go generator to include the referenced package. error="/home/coder/coder/codersdk/healthsdk/interfaces.go:6:2: could not import tailscale.com/net/interfaces (invalid package name: \"\")" pkg=github.com/coder/coder/v2/codersdk/healthsdk directory=github.com/coder/coder/v2/codersdk/healthsdk
2024/12/17 15:29:55 ERROR failed to add enum list Experiments: node "Experiments" already exists
2024/12/17 15:29:55 ERROR failed to add enum list Entitlements: node "Entitlements" already exists
Formatted 1 file in 13ms. Fixed 1 file.
npm notice
npm notice New major version of npm available! 10.9.2 -> 11.0.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v11.0.0
npm notice To update run: npm install -g npm@11.0.0
npm notice
+ pnpm install 
Lockfile is up to date, resolution step is skipped
Packages: +1
+
Progress: resolved 1, reused 0, downloaded 1, added 1, done
```

And will eventually work and run. I also had situations where I had to restart another time.

--- a/Makefile
+++ b/Makefile
@@ -659,8 +659,10 @@ examples/examples.gen.json: scripts/examplegen/main.go examples/examples.go $(sh
 	go run ./scripts/examplegen/main.go > examples/examples.gen.json
 
 coderd/rbac/object_gen.go: scripts/typegen/rbacobject.gotmpl scripts/typegen/main.go coderd/rbac/object.go coderd/rbac/policy/policy.go
-	go run scripts/typegen/main.go rbac object > /tmp/object_gen.go
-	mv /tmp/object_gen.go coderd/rbac/object_gen.go
+	tempdir=$(shell mktemp -d /tmp/typegen_rbac_object.XXXXXX)
+	go run ./scripts/typegen/main.go rbac object > "$$tempdir/object_gen.go"
+	mv -v "$$tempdir/object_gen.go" coderd/rbac/object_gen.go
+	rmdir -v "$$tempdir"
 
 codersdk/rbacresources_gen.go: scripts/typegen/codersdk.gotmpl scripts/typegen/main.go coderd/rbac/object.go coderd/rbac/policy/policy.go
 	# Do no overwrite codersdk/rbacresources_gen.go directly, as it would make the file empty, breaking

--- a/Makefile
+++ b/Makefile
@@ -659,7 +659,8 @@ examples/examples.gen.json: scripts/examplegen/main.go examples/examples.go $(sh
 	go run ./scripts/examplegen/main.go > examples/examples.gen.json
 
 coderd/rbac/object_gen.go: scripts/typegen/rbacobject.gotmpl scripts/typegen/main.go coderd/rbac/object.go coderd/rbac/policy/policy.go
-	go run scripts/typegen/main.go rbac object > coderd/rbac/object_gen.go
+	go run scripts/typegen/main.go rbac object > /tmp/object_gen.go
+	mv /tmp/object_gen.go coderd/rbac/object_gen.go
 
 codersdk/rbacresources_gen.go: scripts/typegen/codersdk.gotmpl scripts/typegen/main.go coderd/rbac/object.go coderd/rbac/policy/policy.go
 	# Do no overwrite codersdk/rbacresources_gen.go directly, as it would make the file empty, breaking

